test/support/micro-jest.mjs

```diff
@@ -0,0 +1,250 @@
+          error(
+            `${JSON.stringify(
+              v1
+            )} expected to match ${v2.toString()}, but not matched`
+          );
+        }
+      },
+      notToMatch: function (v2) {
+        // v2 should be regex
+        if (v1.match(v2)) {
+          tests[0].assertFail++;
+          error(
+            `${JSON.stringify(
+              v1
+            )} not expected to match ${v2.toString()}, but matched`
+          );
+        }
+      },
+    });
+  }
+  if (typeof v1 === "function") {
+    matchers = Object.assign(matchers, {
+      toThrow: function (v2) {
+        try {
+          v1();
+          tests[0].assertFail++;
+          error(`function expected to throw error, but not thrown`);
+        } catch (err) {
+          if (typeof v2 === "string") {
+            if (err.message !== v2) {
+              tests[0].assertFail++;
+              error(
+                `function expected to throw error with message "${v2}", but thrown with message "${err.message}"`
+              );
+            }
+          }
+        }
+      },
+      notToThrow: function (v2) {
+        try {
+          v1();
+        } catch (err) {
+          tests[0].assertFail++;
+          error(`function expected not to throw error, but thrown`);
+        }
+      },
+    });
+  }
+  return matchers;
+}
+
+/**
+ * Start all tests. If idx given, only the idx-th test to be performed
+ * @param {number} idx
+ */
+function start(idx) {
+  if (typeof idx === "number") {
+    var _t = tests[idx];
+    tests = [_t];
+  }
+
+  checkTestProgress();
+}
+
+function checkTestProgress() {
+  setTimeout(() => {
+    if (tests.length > 0) {
+      var t = tests[0];
+      if (t.state === 0) {
+        // idle
+        t.state = 1;
+        print(`\r\x1b[97;43m RUNS \x1b[0m ${t.name}`);
+        try {
+          t.fn((err) => {
+            if (err) {
+              t.state = 3;
+              error(err.toString());
+            } else {
+              t.state = t.assertFail === 0 ? 2 : 3;
+            }
+          });
+        } catch (e) {
+          t.state = 3;
+          console.log(e);
+        }
+      } else if (t.state === 2) {
+        // pass
+        print(`\x1b[2K\r\x1b[97;102m PASS \x1b[0m ${t.name}\r\n`);
+        finished.push(tests.shift());
+      } else if (t.state === 3) {
+        // fail
+        print(
+          `\x1b[2K\r\x1b[97;101m FAIL \x1b[0m\x1b[91m ${t.name}\x1b[0m\r\n`
+        );
+        finished.push(tests.shift());
+      }
+
+      checkTestProgress();
+    } else {
+      console.log("");
+      var l = finished.length;
+      var f = finished.filter((ts) => ts.state === 3).length;
+      var p = l - f;
+      console.log(`Tests: ${p} passed, ${f} failed (${l} total)`);
+
+      if (f > 0) {
+        // njs doesn't have a `process.exit` equivalent
+        // So we have to throw here
+        throw "Single test file failed";
+      }
+    }
+    // The njs cli tool does not support non-zero timeouts
+  }, 0);
```

Comment:
**annotation**:
Two things here:
1. njs doesn't have `setInterval` which this script originally used with a delay of 10ms.  Instead I did a recursive invocation with `setTimeout`
2. For reasons I don't currently understand, njs does not support `setTimeout` with a delay greater than `0` in the cli tool.  This could come back to bite us but so far this seems to work fine.